Lightning on x86_64 will crash when importing an empty table via Parquet

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

1. On Amazon Aurora, create an empty table.

    ```sql
    create database account;
    create table audits_log (id int primary key);
    ```

2. Create an AWS Aurora Snapshot (or directly unzip the attached file → [account.audits_log.parquet.zip](https://github.com/pingcap/tidb/files/14952836/account.audits_log.parquet.zip))

3. Import the snapshot using https://docs.pingcap.com/tidb/stable/migrate-aurora-to-tidb

### 2. What did you expect to see? (Required)

Import success

### 3. What did you see instead (Required)

TiDB Lightning panicked with 

```
panic: counter cannot decrease in value

goroutine 1 [running]:
github.com/prometheus/client_golang/prometheus.(*counter).Add(0xc000af4a98?, 0xc000cdc5c0?)
	/go/pkg/mod/github.com/prometheus/client_golang@v1.15.1/prometheus/counter.go:121 +0xf4
github.com/pingcap/tidb/br/pkg/lightning/importer.(*Controller).estimateChunkCountIntoMetrics(0xc000d26e00, {0x5a35368, 0xc0000e4eb0})
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/lightning/importer/import.go:1032 +0x1c2
github.com/pingcap/tidb/br/pkg/lightning/importer.(*Controller).initCheckpoint(0xc000d26e00, {0x5a35368, 0xc0000e4eb0})
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/lightning/importer/import.go:878 +0x165
github.com/pingcap/tidb/br/pkg/lightning/importer.(*Controller).Run(0xc000d26e00, {0x5a35368, 0xc0000e4eb0})
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/lightning/importer/import.go:511 +0x250
github.com/pingcap/tidb/br/pkg/lightning.(*Lightning).run(0xc0006d6b60, {0x5a353a0, 0xc000080098}, 0xc000abe000, 0xc000abc070)
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/lightning/lightning.go:595 +0x12ab
github.com/pingcap/tidb/br/pkg/lightning.(*Lightning).RunOnceWithOptions(0xc0006d6b60, {0x5a353a0, 0xc000080098}, 0xc000abe000, {0x0, 0x0, 0xc000d98880?})
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/lightning/lightning.go:388 +0x2b7
main.main.func2(0xc000d93680, 0x51c7c66?)
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/cmd/tidb-lightning/main.go:94 +0x74
main.main()
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/cmd/tidb-lightning/main.go:95 +0x6b3
```

### 4. What is your TiDB version? (Required)

```
Release Version: v7.1.4
Git Commit Hash: a8b650f8372ede73b49dcc99289828c676075fb6
Git Branch: heads/refs/tags/v7.1.4
Go Version: go1.20.12
UTC Build Time: 2024-03-07 08:16:32
Race Enabled: false
```

CPU: x86_64, OS: Linux

--- a/pkg/lightning/mydump/loader.go
+++ b/pkg/lightning/mydump/loader.go
@@ -826,7 +826,7 @@ func SampleFileCompressRatio(ctx context.Context, fileMeta SourceFileMeta, store
 // SampleParquetDataSize samples the data size of the parquet file.
 func SampleParquetDataSize(ctx context.Context, fileMeta SourceFileMeta, store storage.ExternalStorage) (int64, error) {
 	totalRowCount, err := ReadParquetFileRowCountByFile(ctx, store, fileMeta)
-	if err != nil {
+	if totalRowCount == 0 || err != nil {
 		return 0, err
 	}
 

--- a/pkg/lightning/mydump/loader_test.go
+++ b/pkg/lightning/mydump/loader_test.go
@@ -1108,7 +1108,7 @@ func TestSampleFileCompressRatio(t *testing.T) {
 	require.InDelta(t, ratio, 5000.0/float64(bf.Len()), 1e-5)
 }
 
-func TestSampleParquetDataSize(t *testing.T) {
+func testSampleParquetDataSize(t *testing.T, count int) {
 	s := newTestMydumpLoaderSuite(t)
 	store, err := storage.NewLocalStorage(s.sourceDir)
 	require.NoError(t, err)
@@ -1133,7 +1133,7 @@ func TestSampleParquetDataSize(t *testing.T) {
 	t.Logf("seed: %d. To reproduce the random behaviour, manually set `rand.New(rand.NewSource(seed))`", seed)
 	rnd := rand.New(rand.NewSource(seed))
 	totalRowSize := 0
-	for i := 0; i < 1000; i++ {
+	for i := 0; i < count; i++ {
 		kl := rnd.Intn(20) + 1
 		key := make([]byte, kl)
 		kl, err = rnd.Read(key)
@@ -1165,6 +1165,11 @@ func TestSampleParquetDataSize(t *testing.T) {
 	require.NoError(t, err)
 	// expected error within 10%, so delta = totalRowSize / 10
 	require.InDelta(t, totalRowSize, size, float64(totalRowSize)/10)
+}
+
+func TestSampleParquetDataSize(t *testing.T) {
+	t.Run("count=1000", func(t *testing.T) { testSampleParquetDataSize(t, 1000) })
+	t.Run("count=0", func(t *testing.T) { testSampleParquetDataSize(t, 0) })
 }
 
 func TestSetupOptions(t *testing.T) {

